### PR TITLE
lib/master: fix registering handlers after failover

### PR DESCRIPTION
**Pull request by liuzix**
- Fix the master not being able to receive heartbeats from workers after failover.

### CODE REVIEW

File: lib/worker_manager.go [hunk 1]
```diff
@@ -311,7 +387,7 @@ func (m *workerManagerImpl) HandleHeartbeat(msg *HeartbeatPingMessage, fromNode
 	log.L().Debug("received heartbeat", zap.Any("msg", msg))
 	workerInfo, ok := m.workerInfos[msg.FromWorkerID]
 	if !ok {
-		if m.initialized {
+		if m.fsmState.Load() != workerManagerWaitingHeartbeats {
```
Comment (amyangfei) [inline lib/worker_manager.go:390]: If the `fsmState` has been `workerManagerNormal`, and a worker with the same id restarts, will the worker id in `m.workerInfos`?

Comment (liuzix) [reply lib/worker_manager.go:390]: In the case, the restart must have been triggered by `CreateWorker` on this master, so the worker info is added already.

File: lib/worker_manager.go [hunk 2]
```diff
@@ -173,6 +189,17 @@ func (m *workerManagerImpl) asyncLoadAllWorkers(ctx context.Context) error {
 			// If the worker is proven alive, it will be removed from the
 			// tombstone list.
```
Comment (nongfushanquan) [inline lib/worker_manager.go:190]: a topic for the tombstone worker will be created

File: lib/worker_manager.go [hunk 3]
```diff
@@ -184,6 +211,55 @@ func (m *workerManagerImpl) asyncLoadAllWorkers(ctx context.Context) error {
...
+		return errors.Trace(err)
+	}
+	if !ok {
+		log.L().Panic("duplicate handler",
+			zap.String("topic", topic))
+	}
+	return nil
+}
+
+func (m *workerManagerImpl) unregisterHandler(workerID WorkerID) error {
```
Comment (nongfushanquan) [inline lib/worker_manager.go:245]: it is an ugly implement about the message topic in the master now

**Pull request closed without merging**